server/src/routes/auth/signIn/signIn.js

```diff
@@ -0,0 +1,30 @@
+import bcrypt from 'bcryptjs';
+import User from '../../../models';
+import { signInValidation } from './validateSignIn';
+import { StatusCodes } from 'http-status-codes';
+import jwt from 'jsonwebtoken';
+
+
+/**
+ * @param {import('express').Request} req
+ * @param {import('express').Response} res
+ */
+
+ const signIn = async (req, res) => {
+ const { error } = signInValidation(req.body)
+ if (error) return res.status(StatusCodes.BAD_REQUEST).send(error.details[0].message);
+
+ const user = await User.findOne({ email: req.body.email });
+ if (!user) return res.status(StatusCodes.BAD_REQUEST).send('Email is not found');
+
+ const isPasswordValid = await bcrypt.compare(req.body.password, user.password)
+ if (!isPasswordValid) return res.status(StatusCodes.BAD_REQUEST).send('Password incorrect');
```

Comment:
In case of password or email is invalid we should return uniform message, sth like: "Provided credentials are incorrect". Being verbose about which piece of credential is wrong we give away too much information to the attacker. Somebody might first try to make bunch of requests with random emails to find which one is in the database and then just try popular password cuz he already knows this password is there. If we don't leak information about the email he has to try email/password combination which makes this kind of attack a lot more difficult